sdk/storage/AzureStorageBlob/Source/NetworkingPolicy.swift

```diff
@@ -0,0 +1,58 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCore
+import Foundation
+
+/// Describes the type of network connection
+public enum NetworkType {
+    case unknown
+    case disconnected
+    case wifiOrEthernet
+    case cellular
+}
+
+/// Describes the network state
+public struct NetworkState {
+    var type: NetworkType
+}
+
+public struct TransferNetworkPolicy {
+    /// Permit transfers only on permitted values of `NetworkType`.
+    public let transferOver: [NetworkType]
+
+    public let enableAutoResume: Bool
```

Comment:
I propose we also make this a list of `NetworkType`. It makes sense to me to be able to say "allow transfers over any network type, but only auto-resume them on wifi". I'd also want to make it very clear what "auto-resume" means. These are two different scenarios:

1.) The device goes from disconnected to cellular. In this case, auto-resume would permit the TM to resume the transfer that was stopped when we got disconnected.

2.) The device goes from wifi to cellular. In this case, the TM should /not/ auto-resume because if a transfer were manually paused by the developer, it would be surprising that our handy auto-resume feature would kick in at this time and force it to resume.

We need to make sure this is documented well, and also make sure we account for the difference between these two scenarios in our TM auto-resume logic.

Reply:
In Android, @anuchandy distinguishes between system paused and user paused. I'm okay with this change--you might want it to autoresume on Wifi but not necessarily on cellular, for example--but should we then validate for consistency if you only allow transfer over Wifi but pass both Wifi and cellular to autoresume?

Reply:
We could, but I don't think we necessarily need to. If you allow = [wifi] and autoresume = [wifi, cellular] I think it's pretty clear that on wifi, transfers will be allowed and will auto resume, and on cellular transfers will not be allowed (the auto resume doesn't even come into play)

Reply:
Pushed a change that implements this. I included an assert so that having this scenario will fail during development, but not crash a production app.